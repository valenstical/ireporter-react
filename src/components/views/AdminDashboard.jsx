import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import SummaryBox from './SummaryBox';
import Loader from './Loader';
import TableRow from './TableRow';

/**
 * The admin dashboard component
 * @returns {object} The generated JSX object
 */
export default function AdminDashboard({
  summary, reports
}) {
  const { data } = reports;
  return (
    <main className={`login-wrapper top-space account-wrapper relative bg-grey ${reports.isBusy && 'busy'}`}>
      {/* Pills and Tabs */}
      <section className="heading">
        <div className="wrapper">
          <div>
            <Link to="#">
              Reports (
              {data.length}
              )
            </Link>
          </div>
        </div>
      </section>

      {/* Actual content */}
      <section className="profile-page fill-viewport">
        <div className="wrapper">
          <SummaryBox summary={summary} />

          {/* Table */}
          <div className="card card-sm table-overflow">
            <div className="table-wrapper">
              <table className="table table-striped">
                <thead>
                  <tr>
                    <th>Date</th>
                    <th>Report</th>
                    <th>User</th>
                    <th className="text-center">Status</th>
                  </tr>
                </thead>
                <tbody id="reportsBody">
                  {data.map(report => <TableRow report={report} key={report.id} />)}
                </tbody>
              </table>
            </div>
          </div>


        </div>
      </section>

      <Loader />
    </main>

  );
}

AdminDashboard.propTypes = {
  reports: PropTypes.object.isRequired,
  summary: PropTypes.object.isRequired
};

AdminDashboard.defaultProps = {
};
