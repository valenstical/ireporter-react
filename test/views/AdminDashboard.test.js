import React from 'react';
import { shallow } from 'enzyme';
import AdminDashboard from '../../src/components/views/AdminDashboard';

const props = {
  reports: {
    data: [{
      id: 1
    }]
  },
  summary: {}
};

describe('<AdminDashboard>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<AdminDashboard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when busy', () => {
    props.reports.isBusy = true;
    const wrapper = shallow(<AdminDashboard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
