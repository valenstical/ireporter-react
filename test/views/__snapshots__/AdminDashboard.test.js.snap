// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminDashboard> should render without crashing 1`] = `
<main
  className="login-wrapper top-space account-wrapper relative bg-grey undefined"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
    >
      <div>
        <Link
          replace={false}
          to="#"
        >
          Reports (
          1
          )
        </Link>
      </div>
    </div>
  </section>
  <section
    className="profile-page fill-viewport"
  >
    <div
      className="wrapper"
    >
      <SummaryBox
        summary={Object {}}
      />
      <div
        className="card card-sm table-overflow"
      >
        <div
          className="table-wrapper"
        >
          <table
            className="table table-striped"
          >
            <thead>
              <tr>
                <th>
                  Date
                </th>
                <th>
                  Report
                </th>
                <th>
                  User
                </th>
                <th
                  className="text-center"
                >
                  Status
                </th>
              </tr>
            </thead>
            <tbody
              id="reportsBody"
            >
              <TableRow
                key="1"
                report={
                  Object {
                    "id": 1,
                  }
                }
              />
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </section>
  <Loader />
</main>
`;

exports[`<AdminDashboard> should render without crashing when busy 1`] = `
<main
  className="login-wrapper top-space account-wrapper relative bg-grey busy"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
    >
      <div>
        <Link
          replace={false}
          to="#"
        >
          Reports (
          1
          )
        </Link>
      </div>
    </div>
  </section>
  <section
    className="profile-page fill-viewport"
  >
    <div
      className="wrapper"
    >
      <SummaryBox
        summary={Object {}}
      />
      <div
        className="card card-sm table-overflow"
      >
        <div
          className="table-wrapper"
        >
          <table
            className="table table-striped"
          >
            <thead>
              <tr>
                <th>
                  Date
                </th>
                <th>
                  Report
                </th>
                <th>
                  User
                </th>
                <th
                  className="text-center"
                >
                  Status
                </th>
              </tr>
            </thead>
            <tbody
              id="reportsBody"
            >
              <TableRow
                key="1"
                report={
                  Object {
                    "id": 1,
                  }
                }
              />
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </section>
  <Loader />
</main>
`;
